tests/functional/ingestion/IngestionReader.js

```diff
@@ -103,112 +102,83 @@ describe('ingestion reader tests with mock', function fD() {
-            },
-            next => this.db.createCollection('PENSIEVE', err => {
-                assert.ifError(err);
-                return next();
-            }),
-            next => {
-                this.m = this.db.collection('PENSIEVE');
-                this.m.insertOne(dummyPensieveCredentials, {});
-                return next();
-            },
-            next => {
-                this.m.insertOne({
-                    _id: 'configuration/overlay-version',
-                    value: 6,
-                }, {});
-                return next();
-            },
-            next => {
-                this.m.insertOne(dummySSHKey, {});
-                return next();
-            },
-            next => {
-                const cbOnce = jsutil.once(next);
-                zkClient = new ZookeeperManager('localhost:2181', { autoCreateNamespace: true }, dummyLogger);
-                zkClient.once('error', cbOnce);
-                zkClient.once('ready', () => {
-                    zkClient.removeAllListeners('error');
-                    return cbOnce();
+                    return err ? reject(err) : resolve();
                 });
-            },
-            next => initManagement(testConfig, next),
-            next => {
-                const metadataMock = new MetadataMock();
-                httpServer = http.createServer(
-                    (req, res) => metadataMock.onRequest(req, res))
-                    .listen(testPort);
-                next();
-            }
-        ], done);
+            });
+        } catch (err) {
+            throw err;
+        }
+        producer = new BackbeatProducer({
+            kafka: testConfig.kafka,
+            topic: testConfig.extensions.ingestion.topic,
+        });
+        await new Promise(resolve => producer.once('ready', resolve));
+        await new Promise(resolve => {
+            consumer.connect({ timeout: 1000 }, () => {});
+            consumer.once('ready', resolve);
+        });
+
+        consumer.subscribe([testConfig.extensions.ingestion.topic]);
+        await new Promise(resolve => setTimeout(resolve, 2000));
+        await db.createCollection('PENSIEVE');
+        const collection = db.collection('PENSIEVE');
+        await collection.insertOne(dummyPensieveCredentials);
+        await collection.insertOne({
+            _id: 'configuration/overlay-version',
+            value: 6,
+        });
+        await collection.insertOne(dummySSHKey);
+        zkClient = new ZookeeperManager('localhost:2181', { autoCreateNamespace: true }, dummyLogger);
+        await new Promise((resolve, reject) => {
+            zkClient.once('error', reject);
+            zkClient.once('ready', resolve);
+        });
+        await new Promise((resolve, reject) => {
+            initManagement(testConfig, err => {
+                if (err) {
+                    return reject(err);
+                }
+                return resolve();
+            });
+        });
```

Comment:
best to use `util.promisify()` in such case:

```suggestion
        await util.promisify(initManagement)(testConfig);
```